src/components/SingleReview.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import { fetchSingleReview } from "../utils/api";
+import { useParams } from "react-router-dom";
+import { Votes } from "./Votes";
+import { WriteComment } from "./WriteComment";
+import { CommentCards } from "./CommentCards";
+import { Review } from "./Review";
+
+export const SingleReview = () => {
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [review, setReview] = useState([]);
+  let { review_id } = useParams();
+
+  useEffect(() => {
+    fetchSingleReview(review_id)
+      .then((reviewData) => {
+        setIsLoading(false);
+        setReview(reviewData);
+      })
+      .catch((err) => {
+        setIsError(true);
+        setIsLoading(false);
+      });
+  });
```

Comment:
You're missing the dependancy array in your useEffect which is causing an infinite loop of requests. Add it in :D